python/PiFinder/ui/object_list.py

```diff
@@ -299,33 +335,55 @@ def update(self, force=False):
                 # Offset for title
                 line_pos += 20
 
-                item_name = self.create_name_text(_menu_item)
+                item_name = self.create_shortname_text(_menu_item)
+                item_text = ""
                 if self.current_mode == DisplayModes.LOCATE:
                     item_text = self.create_locate_text(_menu_item)
                 elif self.current_mode == DisplayModes.NAME:
                     item_text = self.create_name_text(_menu_item)
                 elif self.current_mode == DisplayModes.INFO:
                     item_text = self.create_info_text(_menu_item)
 
-                if line_number == 3:
-                    item_line = f"{item_name}{item_text}"
-                else:
-                    item_line = f"{item_name} {item_text}"
-
                 # Type Marker
-                line_bg = 0
-                if line_number == 3:
-                    line_bg = 32
+                line_bg = 32 if is_focus else 0
                 marker = self.get_marker(_menu_item.obj_type, line_color, line_bg)
                 if marker is not None:
                     self.screen.paste(marker, (0, line_pos + 2))
 
+                # calculate start of both pieces of text
+                px_per_char = self.base_px_per_char if not is_focus else self.bold_px_per_char
+                chars_per_line = math.floor(self.display.width/px_per_char)
+                begin_x = 12
+                begin_x2 = begin_x + (len(item_name))*px_per_char
```

Comment:
Here, rather than `px_per_char` you can use `line_font.width` since line_font will be an instance of the PiFinder font class.